msal/src/main/java/com/microsoft/identity/client/internal/controllers/LocalMSALController.java

```diff
@@ -463,14 +468,224 @@ public boolean removeCurrentAccount(RemoveAccountCommandParameters parameters) t
-        // TODO: Placeholder to avoid inheritance error. Will be implemented after Command/Controller level PR in Common
-        return null;
+    public AcquireTokenResult acquireDeviceCodeFlowToken(
+            final AuthorizationResult authorizationResult,
+            final DeviceCodeFlowCommandParameters parameters)
+            throws MsalException, ClientException, IOException, InterruptedException {
+
+        // Logging start of method
+        final String methodName = ":acquireDeviceCodeFlowToken";
+        Logger.verbose(
+                TAG + methodName,
+                "Device Code Flow: Polling for token..."
+        );
+
+        // Start telemetry with LOCAL_DEVICE_CODE_FLOW_POLLING
+        Telemetry.emit(
+                new ApiStartEvent()
+                        .putApiId(TelemetryEventStrings.Api.LOCAL_DEVICE_CODE_FLOW_POLLING)
+        );
+
+        // Create empty AcquireTokenResult object
+        final AcquireTokenResult acquireTokenResult = new AcquireTokenResult();
+
+        // Assign authorization result
+        acquireTokenResult.setAuthorizationResult(authorizationResult);
+
+        // Fetch the Authorization Response
+        final MicrosoftStsAuthorizationResponse authorizationResponse = (MicrosoftStsAuthorizationResponse) authorizationResult.getAuthorizationResponse();
+
+        // Create OAuth2Strategy using commandParameters and strategyParameters
+        final OAuth2StrategyParameters strategyParameters = new OAuth2StrategyParameters();
+        strategyParameters.setContext(parameters.getAndroidApplicationContext());
+
+        final OAuth2Strategy oAuth2Strategy = parameters
+                .getAuthority()
+                .createOAuth2Strategy(strategyParameters);
+
+        // DCF protocol step 2: Poll for token
+        TokenResult tokenResult = null;
+
+        // Create token request outside of loop so it isn't re-created after every loop
+        final MicrosoftStsTokenRequest tokenRequest = (MicrosoftStsTokenRequest) oAuth2Strategy.createTokenRequest(
+                mAuthorizationRequest,
+                authorizationResponse,
+                parameters.getAuthenticationScheme()
+        );
+
+        // Fetch wait interval
+        final int interval = Integer.parseInt(authorizationResponse.getInterval()) * 1000;
+
+        String errorCode = "authorization_pending";
+
+        // Loop to send multiple requests checking for token
+        while (errorCode.equals("authorization_pending")) {
+            errorCode = ""; // Reset error code
+
+            // Execute Token Request
+            tokenResult = oAuth2Strategy.requestToken(tokenRequest);
+
+            if (tokenResult.getErrorResponse() != null) {
+                errorCode = tokenResult.getErrorResponse().getError();
+            }
+
+            if (errorCode.equals("authorization_pending")) {
+                // interval is passed through params
+                Thread.sleep(interval);
+            }
+        }
```

Comment:
`authorization_pending` is declared too many times. It must go in a constant in the appropriate constants file.